crates/rspack_plugin_javascript/src/dependency/esm/provide_dependency.rs

```diff
@@ -38,6 +35,10 @@ impl Dependency for ProvideDependency {
     &self.id
   }
 
+  fn loc(&self) -> Option<String> {
+    Some(self.request.to_string())
```

Comment:
I don't get this. Why do this return a clone of `request`?

Reply:
The loc here is `SyntheticDependencyLocation`, and since the `request` and `SyntheticName` are the same, it has not been repeatedly introduced.

Reply:
`ProvidePlugin` essentially dynamically introduces a `provide_dependency` without a real location.

Reply:
Perhaps we should express it as `ProvidePlugin:request_name`?

Reply:
My negligence, I have always believed that we should use the real location of dependency.

Reply:
> we should use the real location of dependency.

LGTM